### ISSUE TITLE
Put back "Reply to All" as default action when replying

### ISSUE THREAD

**Issue by bootika** [thunderbird/thunderbird-android#7055]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

6.707

### Problem you are trying to solve

In the compose screen, the default action / icon is "Reply" and this inherently leads to broken conversations and out of loop mail participants

### Suggested solution

In the compose screen, please put "Reply to All" as the default action / icon instead of "Reply" as it is now.

Because this leads to a lot of group conversations to be broken and leave participants out of the loop, on important issues. Especialy in case of office or business email accounts.

Or at least make this an option in the "Account Settings" area of the app. And one can decide how he or she wants this to be set

Best would be to have both suggestions accepted :)

Thank you

### Screenshots / Drawings / Technical details

![Screenshot_20230706_190601_K-9 Mail](https://github.com/thundernest/k-9/assets/40858521/e3718bab-bb9a-4fd8-9a8f-32c11a3e5c7d)

**Comment by foxacid** [comment]
Yes please. This has also been discussed in the forum:
https://forum.k9mail.app/t/recover-reply-all-button/

**Comment by farfromrefug** [comment]
same issue here.

**Comment by a-name-not-taken** [comment]
Absolutely not.

Have you never had somebody blast out an email to a mailing list and then for days or weeks after get reply after reply intended only for the original author from people who don't know the difference between
"reply" and "reply-all"?

Don't load that gun.

**Comment by doodhout** [comment]
Then what about leaving it up to the user and make this configurable?

Both sides can come up with arguments against the other choice, but that seems besides the point.

The problem this thread correctly highlights is that the button to reply all has been replaced (from a UI perspective, its location) by the reply button and that's confusing and bad UX practice. Don't suddenly change functionality like that, at least make it optional and distinguish it better from the previous situation.

**Comment by bootika** [comment]
I can't say it must be the d3fault option. But, for sure it should be an option in the Account Settings. For work account it has a lott of sense.

Alexandru Butică
+40 723 626 530
- - - - - - - - - - - - - - - - -
dotdesign.ro

**Comment by foxacid** [comment]
> Absolutely not.
> 
> Have you never had somebody blast out an email to a mailing list and then for days or weeks after get reply after reply intended only for the original author from people who don't know the difference between "reply" and "reply-all"?
> 
> Don't load that gun.

I highly doubt that there are any K9 users don't know the difference between "reply" and "reply-all". This is not Outlook, after all...

**Comment by bootika** [comment]
Thunderbird is the first, most important alternative to Outlook, and perhaps the most popular free e-mail client. So maybe for a lot of people Thunderbird is, or at least must be Outlook..

Personally I use K9 for office e-mail and countless times I left people out of the mailing loop because I instinctively pressed the Reply button instead of the three points, then Reply to all, which is quite unintuitive..

Alexandru Butică
+40 723 626 530
- - - - - - - - - - - - - - - - -
dotdesign.ro

**Comment by bootika** [comment]
100% agree with doodhout.

Alexandru Butică
+40 723 626 530
- - - - - - - - - - - - - - - - -
dotdesign.ro

**Comment by a-name-not-taken** [comment]
> Then what about leaving it up to the user and make this configurable?

Yes, that's one way. Another would be to do what virtually every other email client does and have a separate "Reply" and "Reply-all" options.

**Comment by bootika** [comment]
Correct, both solutions are ok :)

Alexandru Butică
+40 723 626 530
- - - - - - - - - - - - - - - - -
dotdesign.ro

**Comment by a-name-not-taken** [comment]
Thunderbird. Take out the words.

![tbird](https://github.com/thundernest/k-9/assets/33181918/e92063db-fc4f-4719-aca1-fc6151bad235)

**Comment by bootika** [comment]
I said, having both options is also ok. But if only one button slot is available in UI, it should be an account setting which one to use.

Alexandru Butică
+40 723 626 530
- - - - - - - - - - - - - - - - -
dotdesign.ro

**Comment by a-name-not-taken** [comment]
> I said, having both options is also ok. But if only one button slot is available in UI, it should be an account setting which one to use.

Sorry, but a UI is required to support the task st hand, not the opposite. If the needs don't fit the UI design, the UI design is wrong. email has been around for almost 50 years and the best practices for basic functions like this have been thoroughly worked out by millions of users. Reinventing the wheel in the interests of some ill-thought-out "new" idea is not helpful to anyone.

**Comment by SMLunchen** [comment]
Yeah please bring that back, or at least configurable default.
I made a mess in email comms just bc of that silly change.
99% is answering to all that are on the list. About 1% is the original sender.

Thank you.

**Comment by keithzg** [comment]
I reply to emails almost exclusively in my work capacity, almost always to multiple participants where only one or two of them are in the To: field, and as such this has been such a persistent annoyance to me that I have been unable to get myself to switch to K-9 Mail as my daily driver mobile email. K-9 Mail is the only email application I have even tried to use in years that doesn't provide an option to have a Reply All button immediately available in some capacity when reading an email.

**Comment by nylimited** [comment]
I cannot understand why this is **still an open issue**. Of all software, Thunderbird allows this customisation easily. I can set my own primary or default choice of either REPLY or REPLY-ALL which should please everyone. I have not used an email client in some time that did not allow this till I started to play with the new Thunderbird Mobile a.k.a.  K-9 mail. Honestly, the lack of this option and also the lack of inline images will make me stay away from using K-9 as a daily driver even though I use most things Mozilla.   ☹️

**Comment by pszi1ard** [comment]
I agree with OP and the others, I think this email client is simply buggy for not allowing to have "Reply All" as default or just one tap away. A phone is not like a desktop machine where a keyboard shortcut can provide access while leaving the button off of the UI.

**Comment by kewisch** [comment]
I appreciate that folks are passionate about this. I'm acknowledging this is something we should eventually look in to, though it isn't as simple as just replacing the action. I can understand the urge to add further comments in support of this, but please remember this is the development issue tracker with a focus on technical implementation.

**Comment by farfromrefug** [comment]
@kewisch i thank you for answering here. As a dev myself i can understand it. However i think you should explain here the technical issue here just to ease people. Tbh seeing there is a `reply all ` entry in the menu next to the default `reply` button, i dont really see how hard it would be to implement a boolean to put `reply all` as the default button and put `reply` in the menu.

**Comment by NickAndreou** [comment]
It is truly disheartening to see so many people asking for this essential feature only to have a reply from the developer 6 months later saying "eventually". The absence of some basic features like this one keeps me using this app only as a push notification, while doing the actual communication on a computer. As a suggestion maybe you can make the reply button to have  a hold drop down menu with a reply all button?

**Comment by foxacid** [comment]
> 6 months later 

That's 18 months...

**Comment by antviro** [comment]
Any progress on this? It is difficult to understand why the less common option is the default. Moreover , we,Thunderbird users, are used to be able to set it up. I cannot understand why it was removed and it is not back.

**Comment by kewisch** [comment]
**As I mentioned earlier, please don't comment asking for updates or when this will be taken care of.** 

To @farfromrefug's point, the reason is that if we change it, it will be unexpected behavior so there needs to be some sort of migration or visual indicator. Going from Reply to Reply All is much more devastating if you do the wrong thing. You might say just add an option to change it, but we want to aim more for good defaults rather than making everything configurable. So we need to do some research on the best default and then make a call on this. You might say it has been changed without all this before, but that doesn't mean we should just wing it again :-) 

This is not on our roadmap right now, though we may be looking at this as part of #8840 where we're revising the toolbars, which will happen later this year.

**Comment by bootika** [comment]
Well, then just put the option to change between Reply and Reply All in the Account Settings. Leave Reply as the default option, but offer to those who use this e-mail client as a business tool, the option to at least choose. I guess it's a decent request.. :)

**Comment by bootika** [comment]
And there is no "Good Default" in this case. Because businesses need Reply All more often and individuals need Reply more often.
Considering Thunderbird is used more on custom e-mail SMTP servers (less Outlook and Gmail and Yahoo! (and so on) users), I'd rather say that small businesses use more Thunderbird than individuals (because individuals use Gmail or Yahoo! on web interface, usually)

**Comment by antviro** [comment]
I understand @kewisch comment, although it is not possible to understand @bootika 's. I have boths accounts in Thunderbird: my individual account and my corporative account. In both cases, around 95% of the times I reply a mail with some colleagues/friends, I , obviously, use reply all, in the other 5%, 4% are errors because of using the android app, and 1% is truly emails in which I really want to write privately to the sender. I barely can understand other people having other ratios.


I do not understand why we would like to make slower the UI for most of our users, but I won't comment anymore in the thread.

Anyway, this is just a small thing... thanks for the efforts in making this tool and the desktop one!

**Comment by bootika** [comment]
@antviro - You're right, I now assessed all my emails (both corporate and private) and the vast majority of my Sent e-mails have multiple addresants. I guess we're moving towards a "group" kind of conversations era.. So your statistics are right, in my case, as well.

**Comment by NickAndreou** [comment]
It's a tiny button for crying out loud! How difficult it is to add reply all next to reply? Why do you have to consider what ignorant people might do instead if considering what people using your app for business would like to do?

**Comment by ryanleesipes** [comment]
Let's get this behavior fixed, I agree it is annoying. I have made the mistake of removing someone from a conversation a few times because of this. @jbott @Alecaddd

**Comment by huguesdk** [comment]
i’m really in favor of this, too. i’ve experienced the same problems. fyi, reply to all is the default behavior in the gmail android app. i think that we should follow the principle of least surprise and do this.

**Comment by jbott-tbird** [comment]
Just to add clarity on where we are at on this - I am waiting for our Design team to give final guidance on this.  I can see the argument both ways, and wish we had two buttons to give both options - or a user configurable setting.  But, I also want to defer to Design on issues about User Experience.  @laurelterlesky @Alecaddd @ryanleesipes

**Comment by laurelterlesky** [comment]
What is essential, but missing is a bottom app bar, which would include, always, reply, reply all, trash, forward and an overflow button (ellipsis). For the best possible solve and aiming at accessibility, at the top where currently there is a reply button, the overflow menu beside that (ellipsis))would also have reply all (among other options). But the main thing here, is that we're missing a bottom app bar which would make navigation with priority actions much easier for everyone.

**Comment by a-name-not-taken** [comment]
Or on the top, where there's a ton of room to the left of the trash can.

Real estate is not a serious problem for 3 or 4 small icons.

**Comment by jbott-tbird** [comment]
It's the "small" in small icons that makes usability more challenging - especially on a small screen, or with accessibility issues such as trembling hands, or other motor issues - hit area is my primary concern.  We'll have some exploration on the layout options, and figure out how best to approach it, but I do like the multiple options instead of just Reply or Reply All.

**Comment by bootika** [comment]
It only needs to be added as an option, which one is the default one.. one year of discussions for a setting, a choice.. :))

**Comment by bootika** [comment]
3 years, pardon me

**Comment by nylimited** [comment]
> Just to add clarity on where we are at on this - I am waiting for our Design team to give final guidance on this. I can see the argument both ways, and wish we had two buttons to give both options - or a user configurable setting. But, I also want to defer to Design on issues about User Experience. [@laurelterlesky](https://github.com/laurelterlesky) [@Alecaddd](https://github.com/Alecaddd) [@ryanleesipes](https://github.com/ryanleesipes)

With due respect to all involved ...   the design team is asleep on the job.  This issue has been bounced around for *years*.  Yes, years.  I stopped using this app years ago when I found this and one or two other things missing which I felt were deal breakers for me.

It does not take years to add a settings choice in references to show one or the other button. Had that been implemented years ago the design team could have debated how to 'best' implement for the last few years. Seriously, this is a most ridiculous multi-year discussion which reduces usability and doesn't do much for confidence in the designers.

How many other apps can you name that do not have a REPLY ALL option that can be the default?

**Comment by Alecaddd** [comment]
I would like to remind everyone that we have a [Code of Conduct](https://github.com/thunderbird/thunderbird-android/blob/main/CODE_OF_CONDUCT.md) that needs to be respected.

Complaining about a missing features or criticizing people are not acceptable behaviours.

We understand that a feature missing or not behaving as expected can be source of frustrations, but as it can be easily observed in this thread, for as many people that want one thing there's an equal amount of people that want the opposite.

Maintaining an application is hard and we're trying to always do the right thing, sometimes we fail, sometimes it takes us years to do something. Up until last year this code base was maintained by 2 people, servicing an average of 1M active users.

We're trying to improve and do better, let's please remain civil and not pollute the bug tracker.

**Comment by laurelterlesky** [comment]
**Here's the phase 1 solve with a bottom sheet:** 

If a message contains multiple recipients, the reply header button becomes context-aware and opens the bottom sheet, so the user can choose who to reply to.

<img width="1390" height="1060" alt="Image" src="https://github.com/user-attachments/assets/4255bf41-a711-4b0d-98c4-a211621a565b" />

https://zpl.io/AOAY7lO

**Comment by laurelterlesky** [comment]
Also to note, Reply all remains available in the menu overflow (kabob icon) to the right of the reply button in the header.

**Comment by a-name-not-taken** [comment]
That seems WAY too complex.

Not to scale, of course, but why not just add icons for reply, reply all, forward, trash, mark unread, etc., and an expanded menu, like so:

<img width="225" height="69" alt="Image" src="https://github.com/user-attachments/assets/fd8fb5a3-e881-4aef-823d-20267b735fef" />

into the blank space at the top of the screen:

<img width="270" height="555" alt="Image" src="https://github.com/user-attachments/assets/72f2f82d-c17a-40eb-b398-b8917517937a" />

**Comment by kwmbe** [comment]
I agree with what @a-name-not-taken is saying here. While it does look nice, I don't feel like adding an extra step (click reply icon -> choose reply/reply all) is the solution here. However, it's probably fine to keep the reply icon where it is, and I feel like there's enough space to just put a "reply all" icon next to the current reply icon? I edited @a-name-not-taken's picture as an illustration.

<img width="270" height="555" alt="Image" src="https://github.com/user-attachments/assets/cd754318-7258-48ac-b278-ac4f605273f8" />

If there isn't enough space for both icons, one of the icons should be hidden in the kebab like it is now.
However, since it's easier to *remove* people from the recipient list when drafting a mail, than it is to *add back* recipients (they might not be in your contact list, or you copy-paste them from the mail history incorrectly), having "reply all" as the default for mails with more than one recipient feels like the better default.  
This change from reply to reply all could then be communicated by changing the "reply" icon to the double arrow "reply all" icon when there are multiple recipients. The reply option could then be hidden in the kebab.

**Comment by jbott-tbird** [comment]
https://app.zeplin.io/project/69a86394e73ddec6e14d0772/screen/69e26b1d26cfc83958f367b3

**Comment by a-name-not-taken** [comment]
> https://app.zeplin.io/project/69a86394e73ddec6e14d0772/screen/69e26b1d26cfc83958f367b3

Seems to require an account to view that link.

**Comment by Alecaddd** [comment]
The link simply shows the same screenshot that are in this comment: https://github.com/thunderbird/thunderbird-android/issues/7055#issuecomment-4270166920

**Comment by rafaeltonholo** [comment]
FYI: I'm starting to work on this issue.
